src/overlay.jsx

```diff
@@ -44,28 +45,29 @@ const Overlay = React.createClass({
   propTypes: {
     autoLockScrolling: React.PropTypes.bool,
     show: React.PropTypes.bool,
-    transitionEnabled: React.PropTypes.bool,
     style: React.PropTypes.object,
+    transitionEnabled: React.PropTypes.bool,
   },
 
   getDefaultProps() {
     return {
+      show:true,
       autoLockScrolling: true,
       transitionEnabled: true,
+      style:{},
     };
   },
 
   componentDidMount() {
     this._originalBodyOverflow = document.getElementsByTagName('body')[0].style.overflow;
+    if (this.state.show !== this.props.show) {
+      this.setState({show:this.props.show}, this._applyAutoLockScrolling);
```

Comment:
Can we move this to the `componentWillReceiveProps` method.
This will save us one render call.